### PR TITLE
added ReminderCancelled event

### PR DESCRIPTION
**Pull request by Ghostvv**
**Proposed changes**:
- finish #1689

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog

### CODE REVIEW

File: rasa_core/processor.py [hunk 1]
```diff
@@ -336,7 +338,23 @@ def should_predict_another_action(action_name, events):
...
+                             tracker.sender_id)
+
+    @staticmethod
+    async def _cancel_reminders(events: List[Event],
+                                tracker: DialogueStateTracker) -> None:
+        # All Reminders with the same name will be cancelled
+        if events is not None:
+            for e in events:
+                if isinstance(e, ReminderCancelled):
+                    name_to_check = e.action_name +\
```
Comment (tmbo) [inline rasa_core/processor.py]: should use `(...)` instead of `\`

File: rasa_core/processor.py [hunk 2]
```diff
@@ -336,7 +338,23 @@ def should_predict_another_action(action_name, events):
...
-                        name=str(e.action_name))
+                        name=str(e.action_name) +
+                             ACTION_NAME_SENDER_ID_CONNECTOR_STR +
+                             tracker.sender_id)
+
+    @staticmethod
+    async def _cancel_reminders(events: List[Event],
+                                tracker: DialogueStateTracker) -> None:
+        # All Reminders with the same name will be cancelled
+        if events is not None:
```
Comment (tmbo) [inline rasa_core/processor.py]: this can never be `None` according to the type annotation

File: rasa_core/processor.py [hunk 3]
```diff
@@ -336,7 +338,23 @@ def should_predict_another_action(action_name, events):
...
+            for e in events:
+                if isinstance(e, ReminderCancelled):
+                    name_to_check = e.action_name +\
+                                    ACTION_NAME_SENDER_ID_CONNECTOR_STR +\
+                                    tracker.sender_id
+                    for j in (await jobs.scheduler()).get_jobs():
```
Comment (tmbo) [inline rasa_core/processor.py]: the wait is ok, but I'd rather do this once before the loop, e.g.
```
scheduler = await jobs.scheduler()
for j in scheduler.get_jobs():
```

File: tests/test_processor.py [hunk 4]
```diff
@@ -119,6 +118,29 @@ def loop():
...
+    t.update(r)
+    t.update(ReminderCancelled("utter_greet"))
+
+    default_processor.tracker_store.save(t)
+    await default_processor.handle_reminder(r, d)
+
+    # retrieve the updated tracker
+    t = default_processor.tracker_store.retrieve(sender_id)
+    # there should be no utter_greet action
+    assert ActionExecuted("utter_greet") not in t.events
```
Comment (tmbo) [inline tests/test_processor.py]: I think we should also check the jobs of the scheduler here making sure it is not scheduled

File: rasa_core/events/__init__.py [hunk 5]
```diff
@@ -567,6 +567,44 @@ def _from_story_string(
                                   parameters.get("timestamp"))]
 
 
+# noinspection PyProtectedMember
+class ReminderCancelled(Event):
+    """ Cancel all jobs with a specific name."""
```
Comment (tmbo) [inline rasa_core/events/__init__.py]:
```suggestion
    """Cancel all jobs with a specific name."""
```

File: rasa_core/processor.py [hunk 6]
```diff
@@ -336,7 +338,23 @@ def should_predict_another_action(action_name, events):
...
+                        if j.name == name_to_check:
+                            (await jobs.scheduler()).remove_job(j.id)
```
Comment (Ghostvv) [inline rasa_core/processor.py]: @tmbo here `scheduler.remove_job(j.id)` as well?

File: tests/test_processor.py [hunk 7]
```diff
@@ -119,6 +119,53 @@ def loop():
...
+    # check that the jobs were added
+    assert len((await jobs.scheduler()).get_jobs()) == 2
+
+    for t in trackers:
+        await default_processor._cancel_reminders(t.events, t)
+    # check that only one job was removed
+    assert len((await jobs.scheduler()).get_jobs()) == 1
+
+    # # execute the second reminder
+    await default_processor.handle_reminder(reminders[1], dispatchers[1])
```
Comment (Ghostvv) [inline tests/test_processor.py]: @tmbo I don't like the line here

**Pull request merged**